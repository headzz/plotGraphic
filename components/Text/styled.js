import styled from "styled-components";

export const TextStyled = styled.p`
  color: ${(props) => (props.color ? props.color : "inherit")};
  font-family: Lato, Helvetica, sans-serif;
  font-size: ${(props) => (props.large ? "1.5rem" : "1rem")};
  font-weight: ${(props) => (props.bold ? "700" : "400")};
`;
