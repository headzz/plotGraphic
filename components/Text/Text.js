import PropTypes from "prop-types";
import React from "react";
import { TextStyled } from "./styled";
import { BLACK } from "@configuration/constants";

/**
 * @component
 * @name Text
 * @description Responsible for showing styled texts at the screen.
 * @namespace components components/Text
 * @param {string} children The text that will appear inside the component
 * @param {boolean} bold Returns a text with weight bold.
 * @param {string}  color Sets custom color for the text.
 * @param {boolean} large Returns a large size text.
 *
 */
export const Text = ({ bold, children, color, large }) => {
  return (
    <TextStyled bold={bold} color={color} large={large}>
      {children}
    </TextStyled>
  );
};

Text.propTypes = {
  bold: PropTypes.bool,
  children: PropTypes.string.isRequired,
  color: PropTypes.string,
  large: PropTypes.bool,
};

Text.defaultProps = {
  bold: false,
  color: BLACK,
  large: false,
};
