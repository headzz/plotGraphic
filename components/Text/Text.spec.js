import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import { Text } from "./Text";
afterEach(cleanup);

test("render Text content", () => {
  const { container, getByText } = render(<Text>example text</Text>);
  expect(container.firstChild).toBeInTheDocument();
  expect(getByText("example text")).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});

test("render large Text", () => {
  const { container } = render(<Text large>example text</Text>);
  expect(container.firstChild).toHaveStyle("font-size: 1.5rem");
  expect(container.firstChild).toMatchSnapshot();
});

test("render bold Text", () => {
  const { container } = render(<Text bold>example text</Text>);
  expect(container.firstChild).toHaveStyle("font-weight: 700");
  expect(container.firstChild).toMatchSnapshot();
});

test("render Text with another color", () => {
  const { container, getByTestId } = render(
    <Text color="#F0F0F0">example text</Text>
  );
  expect(container.firstChild).toHaveStyle("color: #F0F0F0");
  expect(container.firstChild).toMatchSnapshot();
});
