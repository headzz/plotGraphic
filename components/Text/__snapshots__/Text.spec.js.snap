// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Text content 1`] = `
<p
  class="sc-bdVaJa bCsVaL"
  color="#000000"
>
  example text
</p>
`;

exports[`render Text with another color 1`] = `
<p
  class="sc-bdVaJa jEGDka"
  color="#F0F0F0"
>
  example text
</p>
`;

exports[`render bold Text 1`] = `
<p
  class="sc-bdVaJa cvYyKL"
  color="#000000"
>
  example text
</p>
`;

exports[`render large Text 1`] = `
<p
  class="sc-bdVaJa kohSMR"
  color="#000000"
>
  example text
</p>
`;
